schemas/rule.yml

```diff
@@ -0,0 +1,83 @@
+---
+- name: rule
+  title: Rule
+  group: 2
+  short: Fields to capture details about rules used by observers and agents to generate alerts or other events.
+  description: >
+   Rule fields are used to capture the specifics of any observer or agent rules that generate alerts or other events..
+   Examples of data sources that would populate the rule fields include: platforms, network or host IDS/IPS, network 
+   firewalls, web access firewalls, url filters, endpoint detection and response (EDR) systems, etc.  
+  
+  type: group
+  fields:
+
+    - name: id
+      level: extended
+      type: keyword
+      short: Rule ID
+      description: >
+        A rule ID that is unique within the scope of an agent, observer, or other entity using the rule for detection of this event
+
+      example: 101 
+
+    - name: uuid
+      level: extended
+      type: keyword
+      short: Rule UUID
+      description: >
+        A rule ID that is unique within the scope of a set or group of agents, observers, or other entities using the rule for detection of this event.
+
+      example: 1100110011
+
+    - name: version
+      level: extended
+      type: keyword
+      short: Rule Version
+      description: The version / revision of the rule being used for analysis.
+      example: 1.1
+
+    - name: name
+      level: extended
+      type: keyword
+      short: Rule Name
+      description: The name of the rule generating the event.
+      example: Filter_DOH
+
+
+    - name: description
+      level: extended
+      type: keyword
+      short: Rule Description
+      description: The description of the rule generating the event.
+      example: Block requests to public DNS over HTTPS / TLS protocols
+
+    - name: type
+      level: extended
+      type: keyword
+      short: Rule Type
+      description: >
+         A categorization value keyword used by the entity using the rule for detection of this event
+
+      example: DNS
+      
+    - name: ruleset
```

Comment:
What's the relationship between ruleset and type? Is there a hierarchy there?

Reply:
Ruleset is a logical grouping (e.g. web filter rule group) type is what kind of action does the rule concern itself with, I'm not sure type is necessary across the board.

Reply:
per ecs meeting - I'm thinking this can probably be dropped for now....

Reply:
Ok, I'm good with dropping `.type` for now.

Just to be sure we're on the same page, `.ruleset` is meant to capture a value according to the source of these events?

For example, using this to map Suricata alerts [like this one](https://github.com/elastic/beats/blob/6f14a606f1518a96c4378d3a16a8fd06266a20e7/x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json#L45-L49), we would have this?

`rule.category="Attempted Information Leak"` (from `suricata.eve.alert.category`)

Reply:
Dropped, tho now I am reading the above and thinking maybe we need discrete rulesets (groups of rules/signatures that may apply to e.g. specific networks - e.g. GUEST NET IPS SIGNATURE <rule group> AND rule.type or rule.category - e.g. "Attempted Information Leak" as they are definitely different things logistically

I think category is more descriptive so I will add that in.